src/days-2-4-adv/adv.py

```diff
@@ -33,19 +35,82 @@
 #
-
-# Make a new player object that is currently in the 'outside' room.
-
-# Write a loop that:
-#
-# * Prints the current room name
-# * Prints the current description (the textwrap module might be useful here).
-# * Waits for user input and decides what to do.
+#     'torch': Item("torch",
+#                         """Find your way!"""),
 #
-# If the user enters a cardinal direction, attempt to move to the room there.
-# Print an error message if the movement isn't allowed.
+#     'amulet':   Item("amulet",
+#                          """Stop spells"""),
 #
-# If the user enters "q", quit the game.
+# }
+
+coins = Treasure("coins", "gold coins for the taking")
+rubies = Treasure("rubies", "sparkleys")
+
+sword = Item("sword", "slay the creeps using the sword.")
+ax = Item("ax", "crush them with the ax")
+mace = Item("mace", "fly anything with the mace")
+
+
+
+playerStartingItems = [sword]
+room['outside'].addItem(ax)
+room['foyer'].addItem(mace)
+room['treasure'].addItem(coins)
+room['treasure'].addItem(rubies)
+
+valid_directions = {"n": "n", "s": "s", "e": "e", "w": "w",
+                    "forward": "n", "backwards": "s", "right": "e", "left": "w"}
+
+player = Player(input("What is your name? "), room['outside'],playerStartingItems)
+
+print(" debug string ",player.currentRoom)
+
+while True:
+    print(" debug string ",player)
+    cmds = input("-> ").lower().split(" ")
+    if len(cmds) == 1:
+        if cmds[0] == "q":
+            break
+        elif cmds[0] in valid_directions:
+            player.travel(valid_directions[cmds[0]])
+        elif cmds[0] == "look":
+            player.look()
+        elif cmds[0] == "i" or cmds[0] == "inventory":
+            player.printInventory()
+        elif cmds[0] == "status":
+            player.printStatus()
+        else:
+            print("I did not understand that command.")
+    else:
+        if cmds[0] == "look":
+            if cmds[1] in valid_directions:
+                player.look(valid_directions[cmds[1]])
+        elif cmds[0] == "take" or cmds[0] == "get":
+            itemToTake = player.currentRoom.findItemByName(" ".join(cmds[1:]))
+            if itemToTake is not None:
+                player.addItem(itemToTake)
+                player.currentRoom.removeItem(itemToTake)
+                print(f"You have picked up {itemToTake.name}")
+            else:
+                print("You do not see that item.")
+        elif cmds[0] == "drop":
+            player.dropItem(cmds[1:])
+        elif cmds[0] == "score":
+            items = player.findItemByName(" ".join(cmds[1:]))
+            if items is not None and hasattr(items, "score") and items.score() > 0:
+                strengthGain = int(items.eat() / 10)
+                player.score += score
+                player.item(items)
+                del items
+                print(f"Your score {score} strength!")
+            else:
+                print("You cannot eat that.")
```

Comment:
This looks like it's going to be awesome if you include the ability to eat and track the player's strength. Great idea!